compiler/plc_ast/src/ast.rs

```diff
@@ -576,332 +576,198 @@ pub enum ReferenceAccess {
                 f.debug_struct("DirectAccess").field("access", access).field("index", index).finish()
             }
-            AstStatement::HardwareAccess { direction, access, address, location, .. } => f
+            AstStatement::HardwareAccess(HardwareAccess { direction, access, address }) => f
                 .debug_struct("HardwareAccess")
                 .field("direction", direction)
                 .field("access", access)
                 .field("address", address)
-                .field("location", location)
+                .field("location", &self.location)
                 .finish(),
-            AstStatement::MultipliedStatement { multiplier, element, .. } => f
+            AstStatement::MultipliedStatement(MultipliedStatement { multiplier, element }, ..) => f
                 .debug_struct("MultipliedStatement")
                 .field("multiplier", multiplier)
                 .field("element", element)
                 .finish(),
-            AstStatement::CaseCondition { condition, .. } => {
+            AstStatement::CaseCondition(condition) => {
                 f.debug_struct("CaseCondition").field("condition", condition).finish()
             }
-            AstStatement::ReturnStatement { .. } => f.debug_struct("ReturnStatement").finish(),
-            AstStatement::ContinueStatement { .. } => f.debug_struct("ContinueStatement").finish(),
-            AstStatement::ExitStatement { .. } => f.debug_struct("ExitStatement").finish(),
-            AstStatement::CastStatement { target, type_name, .. } => {
+            AstStatement::ReturnStatement(..) => f.debug_struct("ReturnStatement").finish(),
+            AstStatement::ContinueStatement(..) => f.debug_struct("ContinueStatement").finish(),
+            AstStatement::ExitStatement(..) => f.debug_struct("ExitStatement").finish(),
+            AstStatement::CastStatement(CastStatement { target, type_name }) => {
                 f.debug_struct("CastStatement").field("type_name", type_name).field("target", target).finish()
             }
-            AstStatement::ReferenceExpr { access, base, .. } => {
+            AstStatement::ReferenceExpr(ReferenceExpr { access, base }) => {
                 f.debug_struct("ReferenceExpr").field("kind", access).field("base", base).finish()
             }
         }
     }
 }
 
-impl AstStatement {
+impl AstNode {
     ///Returns the statement in a singleton list, or the contained statements if the statement is already a list
-    pub fn get_as_list(&self) -> Vec<&AstStatement> {
-        if let AstStatement::ExpressionList { expressions, .. } = self {
-            expressions.iter().collect::<Vec<&AstStatement>>()
+    pub fn get_as_list(&self) -> Vec<&AstNode> {
+        if let AstStatement::ExpressionList(expressions) = &self.stmt {
+            expressions.iter().collect::<Vec<&AstNode>>()
         } else {
             vec![self]
         }
     }
+
     pub fn get_location(&self) -> SourceLocation {
```

Comment:
Should we possibly make this a &SourceLocation? we could leave this for a later commit, but it kind of makes sense for a getter maybe?

Reply:
I'll open an issue for it  so we can submit this PR faster